packages/docs/src/routes/demo/cookbook/portal/portal-provider.tsx

```diff
@@ -0,0 +1,105 @@
+import {
+  $,
+  Slot,
+  component$,
+  createContextId,
+  useContext,
+  useContextProvider,
+  useSignal,
+  useStylesScoped$,
+  type ContextId,
+  type QRL,
+  type Signal,
+} from '@builder.io/qwik';
+import { type JSXNode } from '@builder.io/qwik/jsx-runtime';
+import CSS from './portal-provider.css?inline';
+
+// Define public API for opening up Portals
+export const PortalAPI = createContextId<
```

Comment:
I think we should rename it to `PortalAPIContextId` to make it different than the actual API